pkg/download/download.go

```diff
@@ -75,7 +77,25 @@ func (p *protocol) request(mod, ver string) error {
 }
 
 func (p *protocol) List(ctx context.Context, mod string) ([]string, error) {
-	return p.dp.List(ctx, mod)
+	const op errors.Op = "protocol.List"
+	goList, goErr := p.dp.List(ctx, mod)
+	strList, sErr := p.s.List(ctx, mod)
+	if goErr != nil && sErr != nil {
+		var errs error
+		errs = multierror.Append(errs, goErr, sErr)
+		return nil, errors.E(op, errs)
+	}
+	if goErr != nil && sErr == nil {
+		semver.SortVersions(strList)
```

Comment:
It's not guaranteed that storage returns a sorted versions list. Should storage.List take care of this? What do you think?

Reply:
@marpio I think Go does not require us to sort versions. Also, I'm maintaining a copy of the Go internal semver stuff here github.com/marwan-at-work/vgop (including parse of go.mod file etc) but I don't think we need to do any sorting any way.